### ISSUE TITLE
完全按照demo写的，成功连接socket服务后就会报下面的错

### ISSUE THREAD

**Issue by rellyCheng** [infitio/flutter_socket_io#19]
E/EventThread(17866): Task threw exception
E/EventThread(17866): java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread. Current thread: EventThread
E/EventThread(17866): 	at io.flutter.embedding.engine.FlutterJNI.ensureRunningOnMainThread(FlutterJNI.java:605)
E/EventThread(17866): 	at io.flutter.embedding.engine.FlutterJNI.dispatchPlatformMessage(FlutterJNI.java:515)
E/EventThread(17866): 	at io.flutter.embedding.engine.dart.DartMessenger.send(DartMessenger.java:74)
E/EventThread(17866): 	at io.flutter.embedding.engine.dart.DartExecutor.send(DartExecutor.java:163)
E/EventThread(17866): 	at io.flutter.view.FlutterNativeView.send(FlutterNativeView.java:152)
E/EventThread(17866): 	at io.flutter.plugin.common.MethodChannel.invokeMethod(MethodChannel.java:95)
E/EventThread(17866): 	at io.flutter.plugin.common.MethodChannel.invokeMethod(MethodChannel.java:82)
E/EventThread(17866): 	at com.infitio.adharasocketio.AdharaSocket$1.call(AdharaSocket.java:81)
E/EventThread(17866): 	at io.socket.emitter.Emitter.emit(Emitter.java:117)
E/EventThread(17866): 	at io.socket.client.Socket.access$601(Socket.java:24)
E/EventThread(17866): 	at io.socket.client.Socket$5.run(Socket.java:186)
E/EventThread(17866): 	at io.socket.thread.EventThread.exec(EventThread.java:55)
E/EventThread(17866): 	at io.socket.client.Socket.emit(Socket.java:182)
E/EventThread(17866): 	at io.socket.client.Socket.onconnect(Socket.java:389)
E/EventThread(17866): 	at io.socket.client.Socket.onpacket(Socket.java:283)
E/EventThread(17866): 	at io.socket.client.Socket.access$100(Socket.java:24)
E/EventThread(17866): 	at io.socket.client.Socket$2$2.call(Socket.java:120)
E/EventThread(17866): 	at io.socket.emitter.Emitter.emit(Emitter.java:117)
E/EventThread(17866): 	at io.socket.client.Manager.ondecoded(Manager.java:427)
E/EventThread(17866): 	at io.socket.client.Manager.access$1600(Manager.java:30)
E/EventThread(17866): 	at io.socket.client.Manager$7.call(Manager.java:403)
E/EventThread(17866): 	at io.socket.parser.IOParser$Decoder.add(IOParser.java:105)
E/EventThread(17866): 	at io.socket.client.Manager.ondata(Manager.java:419)
E/EventThread(17866): 	at io.socket.client.Manager.access$1000(Manager.java:30)
E/EventThread(17866): 	at io.socket.client.Manager$2.call(Manager.java:370)
E/EventThread(17866): 	at io.socket.emitter.Emitter.emit(Emitter.java:117)
E/EventThread(17866): 	at io.socket.engineio.client.Socket.onPacket(Socket.java:551)
E/EventThread(17866): 	at io.socket.engineio.client.Socket.access$1000(Socket.java:36)
E/EventThread(17866): 	at io.socket.engineio.client.Socket$5.call(Socket.java:335)
E/EventThread(17866): 	at io.socket.emitter.Emitter.emit(Emitter.java:117)
E/EventThread(17866): 	at io.socket.engineio.client.Transport.onPacket(Transport.java:126)
E/EventThread(17866): 	at io.socket.engineio.client.transports.Polling.access$700(Polling.java:18)

**Comment by rellyCheng** [comment]
如果注释了红色框内的代码  就不会报异常

**Comment by rellyCheng** [comment]
![1557655660111](https://user-images.githubusercontent.com/22974879/57580809-e41b4700-74e0-11e9-938c-123a2d57cb2f.jpg)

**Comment by rellyCheng** [comment]
已经解决了， 代码没问题 ， 是模拟器的原因，在真机上运行是没问题的。

**Comment by tiholic** [comment]
很高兴你解决了它

**Comment by rinlv** [comment]
I have the same problem with @rellyCheng , this is my dart code:

`import 'package:flutter/material.dart';
import 'package:adhara_socket_io/adhara_socket_io.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  SocketIOManager socketManager;

  @override
  void initState() {
    super.initState();
    socketManager = SocketIOManager();
    initSocket();
  }

  void initSocket() async {
    SocketIO socket =
        await socketManager.createInstance('http://192.168.1.32:5003/');
    socket.onConnect((data) {
      print("connected...");
      print(data);
      socket.emit("message", ["Hello world!"]);
    });
    socket.on("news", (data) {
      //sample event
      print("news");
      print(data);
    });
    socket.connect();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'You have pushed the button this many times:',
            )
          ],
        ),
      )
    );
  }
}
`

**Comment by rinlv** [comment]
I have been run your example and got the same error as @rellyCheng . Did I wrong implementation?

**Comment by tiholic** [comment]
@rinlv did you try on an emulator or physical device?

**Comment by rinlv** [comment]
> @rinlv did you try on an emulator or physical device?

@tiholic Both physical device and emulator?

**Comment by rinlv** [comment]
Hi @tiholic, I realize this error occur when I build with the latest Flutter. I build your example with Flutter stable ver 1.2.1, this error was disappeared and I received a notification. Thanks for your supporting.

**Comment by tiholic** [comment]
@rinlv Thanks for the information. Will check the compatibility with latest stable release.

**Comment by lhcdims** [comment]
As at 2019/06/03:

The following error:

```
java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread. Current thread: EventThread
```

appears when using flutter channel master or beta (i.e. flutter version >= 1.6.3)

disappear when using flutter channel stable (i.e. flutter version = 1.5.4)

I found this problem a few days ago when the channel beta was still using flutter version 1.5.8.

**Comment by tiholic** [comment]
@rinlv the issue is resolved by @lhcdims 
Ref: https://github.com/infitio/flutter_socket_io/issues/26

**Comment by tiholic** [comment]
Fixed released in [0.1.11](https://pub.dev/packages/adhara_socket_io/versions/0.1.11)
